test/TestPSScriptFileInfo.Tests.ps1

```
@@ -0,0 +1,75 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Import-Module "$psscriptroot\PSGetTestUtils.psm1" -Force
+
+Describe "Test Test-PSScriptFileInfo" {
+    BeforeAll {
+        $tmpDir1Path = Join-Path -Path $TestDrive -ChildPath "tmpDir1"
+        $tmpDir2Path = Join-Path -Path $TestDrive -ChildPath "tmpDir2"
+        $tmpDir3Path = Join-Path -Path $TestDrive -ChildPath "tmpDir3"
+        $tmpDirPaths = @($tmpDir1Path, $tmpDir2Path, $tmpDir3Path)
+        Get-NewTestDirs($tmpDirPaths)
+
+        # Path to folder, within our test folder, where we store invalid module and script files used for testing
+        $script:testFilesFolderPath = Join-Path $psscriptroot -ChildPath "testFiles"
+
+        # Path to specifically to that invalid test scripts folder
+        $script:testScriptsFolderPath = Join-Path $testFilesFolderPath -ChildPath "testScripts"
+    }
+
+    AfterAll {
+        $tmpDir1Path = Join-Path -Path $TestDrive -ChildPath "tmpDir1"
+        $tmpDir2Path = Join-Path -Path $TestDrive -ChildPath "tmpDir2"
+        $tmpDir3Path = Join-Path -Path $TestDrive -ChildPath "tmpDir3"
+        $tmpDirPaths = @($tmpDir1Path, $tmpDir2Path, $tmpDir3Path)
+        Get-RemoveTestDirs($tmpDirPaths)
+    }
+
+    It "determine script file with minimal required fields as valid" {    
+        $scriptFilePath = Join-Path -Path $tmpDir1Path -ChildPath "testscript.ps1"
+        $scriptDescription = "this is a test script"
+        New-PSScriptFileInfo -FilePath $scriptFilePath -Description $scriptDescription
+        Test-PSScriptFileInfo $scriptFilePath | Should -Be $true
+    }
+
+    It "not determine script file with Author field missing as valid" {
+        $scriptName = "InvalidScriptMissingAuthor.ps1"
+        $scriptFilePath = Join-Path $script:testScriptsFolderPath -ChildPath $scriptName
+
+        Test-PSScriptFileInfo $scriptFilePath | Should -Be $false
+        # TODO: how to test for warnings? (psScriptMissingAuthor)
```

Missing author should throw an error, right (same with version, description, and guid)?

---

Currently Test-PSScriptFileInfo writes a warning with the errors, instead of directly writing errors or throwing them. The cmdlet also returns false. This is because if this cmdlet were run in a pipeline we wouldn't want pipeline execution to stop and thus don't want to write or throw errors, just warnings. This is my understanding, but happy to change it. We can discuss with Paul & Sydney.